test/functionalTest/cases/2560_custom_notification_json/custom_notification_json_replacements_intermediate.test

```diff
@@ -0,0 +1,307 @@
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Custom notification with JSON with replacements in intermediate elements
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart --pretty-print
+
+--SHELL--
+
+#
+# 01. Create subscription for E1 with custom notification with JSON object
+# 02. Create subscription for E1 with custom notification with JSON array
+# 03. GET /v2/subscriptions and see both subscriptions
+# 04. Create E1 entity to trigger both notifications
+# 05. Dump accumulator, see 2 notifications
+#
+
+echo "01. Create subscription for E1 with custom notification with JSON object"
+echo "========================================================================"
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "id" : "E1",
+        "type": "T"
+      }
+    ]
+  },
+  "notification": {
+    "httpCustom": {
+      "url": "http://127.0.0.1:'${LISTENER_PORT}'/notify",
+      "json": {
+        "x": {
+          "x1": "${A}",
+          "x2": "b"
+        },
+        "y": [ "y1", "${B}" ]
+      }
+    }
+  }
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "02. Create subscription for E1 with custom notification with JSON array"
+echo "======================================================================="
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "id" : "E1",
+        "type": "T"
+      }
+    ]
+  },
+  "notification": {
+    "httpCustom": {
+      "url": "http://127.0.0.1:'${LISTENER_PORT}'/notify",
+      "json": [
+        "22",
+         {
+           "x" : [ "${A}", "x2" ],
+           "y" : 3
+         },
+         [ "z1", "${B}" ]
+      ]
+    }
+  }
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "03. GET /v2/subscriptions and see both subscriptions"
+echo "===================================================="
+orionCurl --url /v2/subscriptions
+echo
+echo
+
+
+echo "04. Create E1 entity to trigger both notifications"
+echo "=================================================="
+payload='{
+  "id": "E1",
+  "type": "T",
+  "A": {
+    "value": 10,
+    "type": "Text"
```

Comment:
Should this be type "Number"?

Reply:
Orion doesn't check attribute type except for some special types (e.g. `DateTime`, `geo:json`, etc.) but I'll do the change you propose as that type would be more meaningful.

Reply:
Fixed in [30a0e0b](https://github.com/telefonicaid/fiware-orion/pull/4157/commits/30a0e0b0537766ee79966265965299f1bf737fb6)